cmd/istio-cni/redirect.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2018 Istio authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines the redirect object and operations.
+package main
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	redirectModeREDIRECT         = "REDIRECT"
+	redirectModeTPROXY           = "TPROXY"
+	defaultProxyStatusPort       = "15020"
+	defaultRedirectToPort        = "15001"
+	defaultNoRedirectUID         = "1337"
+	defaultRedirectMode          = redirectModeREDIRECT
+	defaultRedirectIPCidr        = "*"
+	defaultRedirectExcludeIPCidr = ""
+	defaultRedirectExcludePort   = defaultProxyStatusPort
+
+	includeIPCidrsKey = "traffic.sidecar.istio.io/includeOutboundIPRanges"
+	excludeIPCidrsKey = "traffic.sidecar.istio.io/excludeOutboundIPRanges"
+	includePortsKey   = "traffic.sidecar.istio.io/includeInboundPorts"
+	excludePortsKey   = "traffic.sidecar.istio.io/excludeInboundPorts"
+
+	sidecarInterceptModeKey = "sidecar.istio.io/interceptionMode"
+	sidecarPortListKey      = "status.sidecar.istio.io/port"
+	sidecarStatusKey        = "sidecar.istio.io/status"
+)
+
+var (
+	annotationRegistry = map[string]*annotationParam{
+		"inject":         {injectAnnotationKey, "", alwaysValidFunc},
+		"status":         {sidecarStatusKey, "", alwaysValidFunc},
+		"redirectMode":   {sidecarInterceptModeKey, defaultRedirectMode, validateInterceptionMode},
+		"ports":          {sidecarPortListKey, "", validatePortList},
+		"includeIPCidrs": {includeIPCidrsKey, defaultRedirectIPCidr, validateCIDRListWithWildcard},
+		"excludeIPCidrs": {excludeIPCidrsKey, defaultRedirectExcludeIPCidr, validateCIDRList},
+		"includePorts":   {includePortsKey, "", validatePortListWithWildcard},
+		"excludePorts":   {excludePortsKey, defaultRedirectExcludePort, validatePortList},
```

Comment:
Wonder why using short key such as "inject", "status", etc.
Why not just define the map as
 annotationRegistry = map[string]*annotationParams{
injectAnnotationKey: {"", alwaysValidFunc},
......
}

If it's necessary to use the short key, it seems to be appropriate to define them as constants as well.